tests/post/steps/test_example_solution.py

```diff
@@ -0,0 +1,331 @@
+    "environment '{environment}'"))
+def read_solution_deployments(
+    k8s_appsv1_client, value, path, name, environment
+):
+    def _wait_for_deployment():
+        response = get_deployment(k8s_appsv1_client, name, environment)
+        available_replicas = utils.get_dict_element(response, path)
+        assert available_replicas is not None
+
+        assert literal_eval(value) == available_replicas, (
+            "Expected {} replicas for Deployment {}, got {}".format(
+                value, name, context.get('deployment_obj')
+            )
+        )
+    utils.retry(
+        _wait_for_deployment,
+        times=10,
+        wait=5,
+        name="wait for available Deployments"
+    )
+
+
+@then(parsers.parse("we have no Solution '{name}' archive mounted"))
+def no_solution_mountpoint(host, name, k8s_client):
+    with host.sudo():
+        assert name not in host.mount_point.get_mountpoints(), \
+            'Found Solution mountpoint {}'.format(name)
+
+
+@then(parsers.parse("we have no Solution environment '{name}'"))
+def no_solution_environment(k8s_client, name):
+    solution_environment_obj = get_environment(k8s_client, name)
+    assert solution_environment_obj is None
+
+
+@then(parsers.parse("we have no available Solution '{name}'"))
+def no_available_solution(host, name):
+    assert name not in get_available_solution(host)
+
+
+@then(parsers.parse("we have no Solution Configuration file present"))
+def no_solution_config(host, name):
+    with host.sudo():
+        if host.file(SOLUTION_CONFIGURATION_FILE).exists:
+            assert False, \
+                'Solution Configuration file exists at {}'.format(
+                    SOLUTION_CONFIGURATION_FILE
+                )
+
+
+# }}}
+
+
+# Helper {{{
+
+def get_deployment(k8s_appsv1_client, name, namespace):
+    try:
+        response = k8s_appsv1_client.read_namespaced_deployment(
+            name=name, namespace=namespace
+        ).to_dict()
+    except (ApiException) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        raise
+    return response
+
+
+def get_configmap(k8s_client, name, namespace):
+    try:
+        response = k8s_client.read_namespaced_config_map(
+            name, namespace
+        ).to_dict()
+    except Exception as exc:
+        pytest.fail(
+            "Unable to read {} ConfigMap  in namespace {} with error: {!s}"
+            .format(name, namespace, exc)
+        )
+    return response
+
+
+def list_environments(k8s_client):
+    try:
+        response = k8s_client.list_namespace()
+    except Exception as exc:
+        pytest.fail("Unable to read namespaces: {!s}".format(exc))
+    return response.items
+
+
+def get_environment(k8s_client, name):
+    try:
+        response = k8s_client.read_namespace(name)
+    except (ApiException) as exc:
+        if isinstance(exc, ApiException) and exc.status == 404:
+            return None
+        raise
+    return response
```

Comment:
An environment should not be matched on the namespace name but on a specific label `metadata:labels:solutions.metalk8s.scality.com/environment` with the environment name as the value.
So you must use `list_namespace()` with a label selector, should be something like that:
```
ENVIRONMENT_LABEL = "metadata:labels:solutions.metalk8s.scality.com/environment"
response = k8s_client.list_namespace(label_selector='{}={}'.format(ENVIRONMENT_LABEL, name))
```

Reply:
Ok, i will fix this!